src/git.ts

```diff
@@ -112,17 +112,73 @@ export class Git {
     }
   }
 
-  public async cherryPick(commitShas: string[], pwd: string) {
-    const { exitCode } = await this.git(
-      "cherry-pick",
-      ["-x", ...commitShas],
-      pwd,
-    );
-    if (exitCode !== 0) {
+  public async cherryPick(
+    commitShas: string[],
+    conflictResolution: string,
+    pwd: string,
+  ): Promise<string[] | null> {
+    const abortCherryPickAndThrow = async (
+      commitShas: string[],
+      exitCode: number,
+    ) => {
       await this.git("cherry-pick", ["--abort"], pwd);
       throw new Error(
         `'git cherry-pick -x ${commitShas}' failed with exit code ${exitCode}`,
       );
+    };
+
+    if (conflictResolution === `fail`) {
+      const { exitCode } = await this.git(
+        "cherry-pick",
+        ["-x", ...commitShas],
+        pwd,
+      );
+
+      if (exitCode !== 0) {
+        await abortCherryPickAndThrow(commitShas, exitCode);
+      }
+
+      return null;
+    } else {
+      let uncommitedShas: string[] = [...commitShas];
+
+      // Cherry-pick commit one by one.
+      for (const sha of commitShas) {
```

Comment:
🔧 As you're popping the shas from `uncommittedShas` on line 178 anyways, we could also use a while loop rather than a separate for loop, something like: `while(uncommittedShas.length > 0)`. This would highlight the flow control a bit better

Reply:
:+1: Agreed.